drivers/pds/targetcluster/targetcluster.go

```diff
@@ -248,6 +253,69 @@ func (targetCluster *TargetCluster) SetConfig() error {
 	return nil
 }
 
+// RegisterClusterToControlPlane checks and registers the given target cluster to the controlplane
+func (tc *TargetCluster) RegisterClusterToControlPlane(infraParams *parameters.Parameter, tenantId string, installOldVersion bool) error {
+	log.InfoD("Test control plane url connectivity.")
+	var helmChartversion string
+	var components *pdsapi.Components
+	var serviceAccId string
+
+	controlPlaneUrl := infraParams.InfraToTest.ControlPlaneURL
```

Comment:
please check if InitializeApiComponents can be reused here.

Reply:
done